src/FunderMaps.Data/RepositoryBase.cs

```diff
@@ -26,6 +27,112 @@ public virtual async Task<TEntity> AddGetAsync(TEntity entity)
             return await GetByIdAsync(primaryKey);
         }
 
+        /// <summary>
+        ///     Keypair used as cache identifier.
+        /// </summary>
+        protected class KeyPair
+        {
+            /// <summary>
+            ///     Entity hash key.
+            /// </summary>
+            public int EntityKey { get; set; }
+
+            /// <summary>
+            ///     Object hash key.
+            /// </summary>
+            public int Key { get; set; }
+
+            /// <summary>
+            ///     Keypair identifier.
+            /// </summary>
+            /// <remarks>
+            ///     This is very likely to overflow, however the identity shoud
+            ///     still be unique per keypair.
+            /// </remarks>
+            public int KeyPairIdentity => EntityKey + Key;
+        }
+
+        /// <summary>
+        ///     Build entity hash key.
+        /// </summary>
+        protected KeyPair EntityHashKey(object key)
+            => new KeyPair
+            {
+                EntityKey = typeof(TEntity).GetHashCode(),
+                Key = key.GetHashCode(),
+            };
+
+        /// <summary>
+        ///     Set cache item.
+        /// </summary>
+        /// <remarks>
+        ///     Derived repositories can override this call to change cache behavior.
+        /// </remarks>
+        protected virtual void SetCacheItem(KeyPair key, TEntity value, MemoryCacheEntryOptions options)
+            => AppContext.Cache.Set(key.KeyPairIdentity, value, options);
+
+        /// <summary>
+        ///     Unset cache item.
+        /// </summary>
+        /// <remarks>
+        ///     Derived repositories can override this call to change cache behavior.
+        /// </remarks>
+        protected virtual void UnsetCacheItem(KeyPair key)
+            => AppContext.Cache.Remove(key.KeyPairIdentity);
+
+        /// <summary>
+        ///     Get cache item.
+        /// </summary>
+        /// <remarks>
+        ///     Derived repositories can override this call to change cache behavior.
+        /// </remarks>
+        protected virtual bool GetCacheItem(KeyPair key, out TEntity value)
+        {
+            var hasItem = AppContext.Cache.TryGetValue(key.KeyPairIdentity, out value);
+            if (hasItem)
+            {
+                AppContext.Items.Add("cachehit", 1);
+            }
+            return hasItem;
+        }
+
+        /// <summary>
+        ///     Try get entity from cache.
+        /// </summary>
+        protected bool TryGetEntity(TEntityPrimaryKey key, out TEntity value)
+            => GetCacheItem(EntityHashKey(key), out value);
+
+        /// <summary>
+        ///     Cache entity.
+        /// </summary>
+        protected TEntity CacheEntity(TEntityPrimaryKey key, TEntity value)
+        {
+            var options = new MemoryCacheEntryOptions()
+                .SetSlidingExpiration(TimeSpan.FromMinutes(5))
```

Comment:
Do we want this configurable in our options?

Reply:
Yes, but not now